src/main/java/bio/terra/workspace/app/configuration/ApiResourceConfig.java

```diff
@@ -11,4 +17,22 @@
   public void addResourceHandlers(ResourceHandlerRegistry registry) {
     registry.addResourceHandler("/**").addResourceLocations("classpath:/api/");
   }
+
+  @Autowired private SpanCustomizer spanCustomizer;
+
+  @Override
+  public void addInterceptors(InterceptorRegistry registry) {
+    registry.addInterceptor(
+        new HandlerInterceptor() {
+          @Override
+          public boolean preHandle(
+              HttpServletRequest request, HttpServletResponse response, Object handler)
+              throws Exception {
+            spanCustomizer.tag("session-id", request.getSession().getId());
+            // add tags to Stackdriver traces here
+
+            return true;
+          }
+        });
+  }
```

Comment:
creates a parent span for each API call

Reply:
can this use the MDC id instead of/in addition to the session id?